main.go

```diff
@@ -5,45 +5,114 @@ import (
 	"fmt"
 	"log"
 	"net/http"
+	"strings"
+
+	"github.com/google/uuid"
 )
 
-func getPing(w http.ResponseWriter, r *http.Request) {
-	metodo := r.Method
-	fmt.Printf("metodo: %v\n", metodo)
-	if metodo == "GET" {
+type Book struct {
+	ID        uuid.UUID `json:"id"`
+	Name      string    `json:"name"`
+	Price     *float32  `json:"price"`
+	Inventory *int      `json:"inventory"`
+}
+
+var bookslist []Book
+
+func Ping(w http.ResponseWriter, r *http.Request) {
+	method := r.Method
+	if method == http.MethodGet {
 		w.WriteHeader(http.StatusNoContent)
+		return
 	} else {
 		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
 	}
+}
 
+func Books(w http.ResponseWriter, r *http.Request) {
+	method := r.Method
+	switch method {
+	case http.MethodGet:
+		getBooks(w, r)
+		return
+	case http.MethodPost:
+		postBooks(w, r)
+		return
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
 }
 
-func getBooks(w http.ResponseWriter, r *http.Request) {
+func postBooks(w http.ResponseWriter, r *http.Request) {
+	//Read the Json body and save the entry to newBook
+	var newBook Book
+	err := json.NewDecoder(r.Body).Decode(&newBook)
+	if err != nil {
+		log.Println("error:", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
 
-	bookslist := []struct { //Struct created to handle data of the list of books
-		Name      string  `json:"name"`
-		Price     float32 `json:"price"`
-		Inventory int     `json:"inventory"`
-	}{
-		{"Book 1", 30.20, 2},
-		{"Book 2", 20.30, 1},
-		{"Book 3", 32.20, 5},
+	//Verify if that book already exists in the database
+	for i := range bookslist {
+		bookAlreadyExists := strings.EqualFold(bookslist[i].Name, newBook.Name)
+		if bookAlreadyExists {
+			showBook, err := json.Marshal(bookslist[i])
+			if err != nil {
+				log.Println("error:", err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+			warning := fmt.Sprint("Este livro já existe na base de dados:\n" + string(showBook))
+			w.Write([]byte(warning))
+			return
+		}
 	}
 
+	//Verify if the entry is in a valid format
+	blankFields := ""
+	if newBook.Name == "" {
+		blankFields += "Insira um nome para cadastrar o livro.\n"
+	}
+	if newBook.Price == nil {
+		blankFields += "Insira um preço para cadastrar o livro.\n"
+	}
+	if newBook.Inventory == nil {
+		blankFields += "Insira a quantidade deste livro em estoque.\n"
+	}
+	if blankFields != "" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(blankFields))
+		return
+	}
+
+	//Atribute an ID to the entry
+	newBook.ID = uuid.New()
+
+	//Store the book in the database
+	bookslist = append(bookslist, newBook)
+
+	//Return a sucess message
+	w.Write([]byte("Livro adicionado com sucesso."))
```

Comment:
Lets return a http 201 "Created", this way we will have a better semantic in the endpoint response.

Reply:
"Created" is the message to the user, correct?
Until now, I kept the messages to the user in portuguese and the code internal messages in english, considering an app usage by brazilians. Should I change everything to english, including this "created" message?

Reply:
Ahhhh. Now I saw it. The http status name is Created!